semgrep/dependencyparser/parse_lockfile.py

```diff
@@ -64,13 +64,18 @@ def remove_quotes(s: str) -> str:
 
     _comment, all_deps_text = lockfile_text.split("\n\n\n")
     dep_texts = all_deps_text.split("\n\n")
+    if dep_texts == [""]:  # No dependencies
+        yield from []
     for dep_text in dep_texts:
         lines = dep_text.split("\n")
         package_name = extract_yarn_name(lines[0])
         version = remove_quotes(lines[1].split()[1])
-        resolved = remove_trailing_octothorpe(
-            remove_quotes(lines[2].split()[1]).strip()
-        )
+        if len(lines) >= 3 and lines[2].strip().startswith("resolved"):
```

Comment:
I recognize that this is probably common in the parsing code, so not going to nitpick this, but the `len(lines) >= 3 and lines[2]...` and `lines[2].split()[1]` is extremely opaque to me. Would be nicer to wrap this in a function with a descriptive name or add a comment